### PR TITLE
Remove ExtensionSigner

### PR DESCRIPTION
**Pull request by nateReiners**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 1]
```diff
@@ -56,15 +56,14 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
...
+          return await this.signer.accounts;
         }
 
         case 'net_version':
-          return this.signer?.chainId ?? 1;
+          return (await this.signer?.chainId) ?? 1;
         case 'eth_chainId':
-          return hexStringFromNumber(this.signer?.chainId ?? 1);
-
+          return hexStringFromNumber((await this.signer?.chainId) ?? 1);
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:66]: let's revisit all this

signer.accounts or signer.chainId
they are just arbitrary made-up properties we thought could be useful.

provider.request({method: net_version}) doesn't need to be this complex, if we just relay it directly to signer.

**Review by bangtoven — CHANGES_REQUESTED** [review]
let's revisit

**Comment by cb-heimdall** [comment]
Review Error for bangtoven @ 2024-07-25 23:02:51 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

File: packages/wallet-sdk/src/sign/interface.ts [hunk 2]
```diff
@@ -2,8 +2,8 @@ import { RequestArguments } from ':core/provider/interface';
 import { AddressString } from ':core/type';
 
 export interface Signer {
-  readonly accounts: AddressString[];
-  readonly chainId: number;
+  readonly accounts: AddressString[] | Promise<AddressString[]>;
+  readonly chainId: number | Promise<number>;
```
[the remark this answers is not in the document]
Comment (nateReiners) [reply packages/wallet-sdk/src/sign/interface.ts:6]: yeah the other option is to make the other getters async as well to get rid of the OR type

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts (hunk 1 above)
Comment (nateReiners) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:66]: yeah, I think we can just remove net_version and eth_chainId from here and those requests will be passed to signer.request instead

Comment (nateReiners) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts]: [c627d48](https://github.com/coinbase/coinbase-wallet-sdk/pull/1353/commits/c627d48c5092f1b30c54815b48e8d2d500b5b934)

**Review by bangtoven — APPROVED** [review]
ty @nateReiners 
this reminds me, let's revert the makeProvider changes and make it identical to the public version we've released. will file ticket for this. let's tackle it next week.

**Pull request merged**